### PR TITLE
WIP: update for sdk2 libs. need to fix kv test

### PR DESCRIPTION
**Pull request by ebuchman**
replaces https://github.com/tendermint/tendermint/pull/1070

Changed branch name for consistency

### CODE REVIEW

File: glide.yaml [hunk 1]
```diff
@@ -18,23 +18,23 @@ import:
 - package: github.com/spf13/viper
   version: v1.0.0
 - package: github.com/tendermint/abci
-  version: v0.9.0
+  version: sdk2
```
Comment (odeke-em) [inline glide.yaml]: Am assuming these will be updated later on?

Comment (melekes) [reply glide.yaml]: yes

**Comment by melekes** [comment]
```
BenchmarkSwitchBroadcast-2         30000             71275 ns/op
--- BENCH: BenchmarkSwitchBroadcast-2
        switch_test.go:339: success: 1, failure: 0
        switch_test.go:339: success: 100, failure: 0
        switch_test.go:339: success: 10000, failure: 0
        switch_test.go:339: success: 30000, failure: 0
```

Tested code:

```
func (sw *Switch) Broadcast(chID byte, msg interface{}) chan bool {
	successChan := make(chan bool, len(sw.peers.List()))
	sw.Logger.Debug("Broadcast", "channel", chID, "msg", msg)
	var wg sync.WaitGroup
	for _, peer := range sw.peers.List() {
		wg.Add(1)
		go func(peer Peer) {
			defer wg.Done()
			success := peer.Send(chID, msg)
			successChan <- success
		}(peer)
	}
	go func() {
		wg.Wait()
		close(successChan)
	}()
	return successChan
}
```

**Comment by melekes** [comment]
@ebuchman want me to rebase this against latest develop?

**Comment by ebuchman** [comment]
Probably need to try this again from scratch, not sure if its even worth rebasing. Let's get the new sdk2 branches in the dependencies merged in to their develops and then try again :(

**Pull request closed without merging**